activerecord/test/cases/relation_test.rb

```diff
@@ -307,6 +307,63 @@ def test_relation_merging_keeps_joining_order
       assert_equal 3, ratings.count
     end
 
+    def test_relation_with_annotation_includes_comment_in_to_sql
+      post_with_annotation = Post.where(id: 1).annotate("foo")
+      assert_match %r{= 1 /\* foo \*/}, post_with_annotation.to_sql
+    end
+
+    def test_relation_with_annotation_includes_comment_in_sql
+      post_with_annotation = Post.where(id: 1).annotate("foo")
+      assert_sql(%r{/\* foo \*/}) do
+        assert post_with_annotation.first, "record should be found"
+      end
+    end
+
+    def test_relation_with_annotation_accepts_multiple_comments
+      post_with_annotation = Post.where(id: 1).annotate("foo", "bar")
+      assert_sql(%r{/\* foo bar \*/}) do
+        assert post_with_annotation.first, "record should be found"
+      end
+    end
```

Comment:
Wrapping my head around annotate with multiple arguments. Joining with a space in the SQL statement feels surprising, and I'm not sure what's meant by more than one annotation. For multivalue, I'd imagine we'd want a standard format and key/value pairs rather than arbitrary strings.

Reply:
> Joining with a space in the SQL statement feels surprising, and I'm not sure what's meant by more than one annotation.

Yeah, that's a great point. I suppose I've done this because I could, not because I should. I'll walk it back.

> For multivalue, I'd imagine we'd want a standard format and key/value pairs rather than arbitrary strings.

Yeah, I'd considered this, i.e., accepting a hash and generating `"key1:val1,key2:val2"` strings, but that also seems like something that can be handled as per developer preferences at the application level.